service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractStatusService.java

```diff
@@ -200,6 +215,26 @@ public EhrExtractStatus updateEhrExtractStatusAccessDocumentDocumentReferences(
         return ehrExtractStatus;
     }
 
+    public void updateEhrExtractStatusAcknowledgement(SendAcknowledgementTaskDefinition sendAcknowledgementTaskDefinition,
+        String positiveAckMessageId) {
+        Query query = createQueryForConversationId(sendAcknowledgementTaskDefinition.getConversationId());
+
+        Update update = createUpdateWithUpdatedAt();
+        update.set(ACK_TASK_ID_PATH, sendAcknowledgementTaskDefinition.getTaskId());
+        update.set(ACK_MESSAGE_ID_PATH, positiveAckMessageId);
+        update.set(ACK_TYPE_CODE_PATH, sendAcknowledgementTaskDefinition.getTypeCode());
+
+        sendAcknowledgementTaskDefinition.getReasonCode().ifPresent(reason -> update.set(ACK_REASON_CODE_PATH, reason));
+        sendAcknowledgementTaskDefinition.getDetail().ifPresent(detail -> update.set(ACK_DETAIL_CODE_PATH, detail));
+
+        UpdateResult updateResult = mongoTemplate.updateFirst(query, update, EhrExtractStatus.class);
+
+        if (updateResult.getModifiedCount() != 1) {
+            throw new EhrExtractException("EHR Extract Status was not updated with Acknowledgement Message.");
+        }
+        LOGGER.info("Database successfully updated with Acknowledgement");
```

Comment:
This could be more specific as we will both send and receive acknowledgments

```suggestion
        LOGGER.info("Database updated for sending application acknowledgement");
```